Add MTU to connection status

bboreham opened the pull request:
Fixes #2389 

I carry the data through Mesh in a map to make it generic; AWSVPC declines to supply it in this implementation.

Sample output:

````
$ weave status connections
<- 192.168.48.1:59366    established sleeve 6a:41:53:e1:d7:29(vagrant-ubuntu-wily-64) mtu=1438
<- 192.168.48.11:42128   established fastdp 5e:7e:09:ac:95:94(host1) mtu=1410
$ weave report
...
        "Connections": [
            {
                "Address": "192.168.48.1:59366",
                "Outbound": false,
                "State": "established",
                "Info": "sleeve 6a:41:53:e1:d7:29(vagrant-ubuntu-wily-64)",
                "Data": {
                    "mtu": 1438
                }
            },
            {
                "Address": "192.168.48.11:42128",
                "Outbound": false,
                "State": "established",
                "Info": "fastdp 5e:7e:09:ac:95:94(host1)",
                "Data": {
                    "mtu": 1410
                }
            }
        ],
...
````

brb commented:
Left some comments at https://github.com/weaveworks/mesh/issues/64

brb commented:
@bboreham would you mind adding a test case to `700_status_and_report_test.sh`?

rade commented:
`GetDisplayNameAndAttrs()` is an odd function name and has an odd signature. Wouldn't it be better to either include the display name as an attribute (so we would only have `GetAttrs()`, or rather `Attrs()`), or have two functions?

bboreham commented:
@rade two functions were rejected as a source of races.  Name as an attribute is plausible; wdyt @brb ?

brb commented:
I fine with name as an attribute.

bboreham commented:
I changed the method to `Attrs()` returning just a map, and adjusted all uses to assume the name is under `"name"`.

I also added a test case; note the test is renamed because it requires two hosts now.

brb commented on test/700_status_and_report_2_test.sh:
```diff
@@ -27,6 +28,9 @@ assert_raises "weave_on $HOST1 report | grep nicknamington"
 assert "weave_on $HOST1 status connections | tr -s ' ' | cut -d ' ' -f 2" "10.2.2.1:6783"
+weave_on $HOST1 connect $HOST2
+sleep 1
+assert "weave_on $HOST1 status connections | grep established | tr -s ' ' | cut -d ' ' -f 6" "mtu=1410"
```
Nitpicking: I'd replace `tr` + `cut` with `grep -q`.

brb commented:
* I've merged the PR in `mesh`. You can update the vendored dep.
* I think we should update docs. In particular: https://www.weave.works/docs/net/latest/troubleshooting/#weave-status-connections

bboreham commented:
I updated the test again; PTAL

awh commented:
#### Documentation changes:
modified [troubleshooting](https://dev-weavewww.pantheonsite.io/docs/net/2389-report-mtu/troubleshooting/)
